src/scripts/modules/warningManager.ts

```diff
@@ -0,0 +1,69 @@
+import warningObj from './inputWarning';
+import * as validationMsg from './validationMsg';
+
+/**
+ * @readonly
+ * @enum {InputType}
+ */
+
+enum InputType {
+  EMAIL = 'email',
+  PW = 'pw',
+  PW_CHECK = 'pw-check',
+  NICKNAME = 'nickname',
+}
+
+/**
+ * @function attatchInputWarnMsg
+ * @description input 요소에 상태에 따른 경고 메세지가 출력되는 기능을 부착한다.
+ * @param {HTMLInputElement} input 경고 메세지 출력기능이 부착될 input 요소
+ * @param {InputType} inputType input 요소의 inputType
+ * @param {String} eventName 문자열로 표기된 이벤트 종류
+ * @param {HTMLElement} position 경고 메세지 태그 요소가 추가될 위치(이 요소 의 바로 다음 형제요소로 추가된다.)
+ * @param {Function} checkFunc 유효성 검사에 추가 로직이 필요한 경우 사용되는 함수
+ */
+
+function attatchInputWarnMsg(
+  input: HTMLInputElement,
+  inputType: InputType,
+  eventName: string,
+  position: HTMLElement | null = null,
+  checkFunc: (() => boolean) | null = null
+) {
+  let getvalidateMsg = null;
+  switch (inputType) {
+    case InputType.EMAIL:
+      getvalidateMsg = validationMsg.getEmailValidateMsg;
+      break;
+    case InputType.PW:
+      getvalidateMsg = validationMsg.getPwValidateMsg;
+      break;
+    case InputType.PW_CHECK:
+      getvalidateMsg = validationMsg.getPwCheckValidateMsg;
+      break;
+    case InputType.NICKNAME:
+      getvalidateMsg = validationMsg.getNicknameValidateMsg;
+      break;
+    default:
+      throw new Error('InputType이 잘못 되었습니다.');
+  }
```

Comment:
input 컴포넌트의 타입별로 분기처리하여 메세지를 붙여줄 때 switch 문을 사용해주신 부분 좋습니다. 👍
if처럼 분기마다 조건이 자유로운 것이 아닌 inputType의 값에 따라서 조건을 나누는 것이 분명하게 보이기 때문입니다. 

다만 
- getvalidateMsg를 미리 null로 선언해 두어야하는 부분
- 매 분기별로 break를 설정해야 하는 부분

은 switch문의 약간 아쉬운 부분이기도 합니다.

아래처럼 객체를 만들고 그 객체 안에서 해당하는 키에 따라 값을 가져오는 방식이 조금 더 개발자 입장에서 편리한 부분이 있어 추천드립니다. 😄 

```suggestion
  // attatchInputWarnMsg 함수 밖에 선언
const validateMsgMap = {
  [InputType.EMAIL]: validationMsg.getEmailValidateMsg,
  [InputType.PW]: validationMsg.getPwValidateMsg,
  [InputType.PW_CHECK]: validationMsg.getPwCheckValidateMsg,
  [InputType.NICKNAME]: validationMsg.getNicknameValidateMsg,
};

  
  // ...

  const getvalidateMsg = validateMsgMap[inputType];
  
  if (!getvalidateMsg) {
    throw new Error('InputType이 잘못 되었습니다.');
  }

```